isolates/bin/long_running_isolate.dart

```diff
@@ -0,0 +1,88 @@
+// Spawn an isolate, read multiple files, send their contents to the spawned
+// isolate, and wait for the parsed JSON.
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+final files = [
+  File('json_01.json'),
+  File('json_02.json'),
+  File('json_03.json'),
+];
+
+// Reads multiple files and parses them on a separate isolate.
+Future<void> main() async {
+  // Read some data.
+  final fileData = await _readFiles();
+
+  // Use that data.
+  await for (final jsonData in _parseJsonStrings(fileData)) {
+    print('Received JSON with ${jsonData.length} keys');
+  }
+}
+
+// Reads multiple files simultaneously using Future.wait().
+Future<List<String>> _readFiles() async {
+  return await Future.wait([
+    ...files.map((file) => file.readAsString()),
+  ]);
+}
+
+// Sends multiple JSON strings to a separate isolate.
+Stream<Map<String, dynamic>> _parseJsonStrings(
+    List<String> jsonStrings) async* {
+  final p = ReceivePort();
+  await Isolate.spawn(_receiveAndParseJsonService, p.sendPort);
+
+  // Send each JSON string in order, waiting for the previous result before
+  // sending the next JSON string.
+  var i = 0;
+
+  // The port used to send JSON strings to be parsed.
+  // This is the first message sent by the spawned isolate.
+  late final SendPort sendPort;
+
+  // Receive messages from the spawned isolate
+  await for (final message in p) {
+    if (message is SendPort) {
+      // This is the first message, assign the sendPort
+      sendPort = message;
+    } else if (message is Map<String, dynamic>) {
+      // This is a response, add it to the Stream returned by this function.
+      yield message;
+    }
+
+    // Send the next JSON string to be parsed.
+    if (i < jsonStrings.length) {
+      sendPort.send(jsonStrings[i++]);
```

Comment:
I think this inline postfix increment can cause some temporary confusion for some developers reading this. What do you think about splitting it?

```suggestion
      sendPort.send(jsonStrings[i]);
      i++;
```

or

```suggestion
      sendPort.send(jsonStrings[i]);
      i += 1;
```

Reply:
Both seem okay, but I went with Lasse's `StreamQueue` idea instead.